### PR TITLE
gui: When private key is disabled, only show watch-only balance

### PR DESCRIPTION
**Pull request by ken2812221**
If a wallet is in private key disabled mode, the spendable balance is always zero, it does not have to show on GUI. Show the watch-only balance at normal balance column if a wallet is in that mode.

image

### CODE REVIEW

File: src/qt/overviewpage.cpp [hunk 1]
```diff
@@ -184,13 +184,24 @@ void OverviewPage::setBalance(const interfaces::WalletBalances& balances)
 // show/hide watch-only labels
 void OverviewPage::updateWatchOnlyLabels(bool showWatchOnly)
```
Comment (promag) [inline src/qt/overviewpage.cpp]: nit, could remove this method and below do
```cpp
connect(model, &WalletModel::notifyWatchonlyChanged, [this](bool showWatchOnly) {
    updateWalletLabels(showWatchOnly, !walletModel->privateKeysDisabled());
});
```

**Review by promag — COMMENTED** [review]
Concept ACK. Could add before and after shots.

**Comment by promag** [comment]
With multiple wallets loaded, it may be weird to have the toggling, maybe disable instead of hiding?

**Comment by laanwj** [comment]
Concept ACK

**Comment by jonasschnelli** [comment]
Nice!
Concept ACK

**Comment by ken2812221** [comment]
Rebased

File: src/qt/overviewpage.cpp [hunk 2]
```diff
@@ -161,15 +161,22 @@ void OverviewPage::setBalance(const interfaces::WalletBalances& balances)
...
-    ui->labelWatchImmature->setText(BitcoinUnits::formatWithUnit(unit, balances.immature_watch_only_balance, false, BitcoinUnits::separatorAlways));
-    ui->labelWatchTotal->setText(BitcoinUnits::formatWithUnit(unit, balances.watch_only_balance + balances.unconfirmed_watch_only_balance + balances.immature_watch_only_balance, false, BitcoinUnits::separatorAlways));
-
+    if (walletModel->privateKeysDisabled()) {
+        ui->labelBalance->setText(BitcoinUnits::formatWithUnit(unit, balances.watch_only_balance, false, BitcoinUnits::separatorAlways));
+        ui->labelUnconfirmed->setText(BitcoinUnits::formatWithUnit(unit, balances.unconfirmed_watch_only_balance, false, BitcoinUnits::separatorAlways));
+        ui->labelImmature->setText(BitcoinUnits::formatWithUnit(unit, balances.immature_watch_only_balance, false, BitcoinUnits::separatorAlways));
+        ui->labelTotal->setText(BitcoinUnits::formatWithUnit(unit, balances.watch_only_balance + balances.unconfirmed_watch_only_balance + balances.immature_watch_only_balance, false, BitcoinUnits::separatorAlways));
+    }
```
Comment (laanwj) [inline src/qt/overviewpage.cpp]: code style nit:
`} else {`

Comment (ken2812221) [reply src/qt/overviewpage.cpp]: fixed

**Comment by fanquake** [comment]
Testing https://github.com/bitcoin/bitcoin/pull/13966/commits/fe1ff5026bace9b43ac01c14a26ed0dd75ccc6f7 using the following commands (in the Debug window given that we're looking at a GUI change). Screenshots below, will review further shortly.
```
src/qt/bitcoin-qt --testnet
createwallet "regular-wallet" false
createwallet "watch-only-wallet" true

// import a random testnet address
importaddress "some-address" "random-addr" false false
```
regular-wallet

watch-only

**Comment by jonasschnelli** [comment]
Tested ACK fe1ff5026bace9b43ac01c14a26ed0dd75ccc6f7
Works as expected (also tested with multiwallet)

**Comment by Sjors** [comment]
Concept ACK, the distinction between"spendable" and "watch-only" is confusing anyway, and a big source of [scams](https://blog.blockchain.com/2018/04/11/addressing-watch-only-scams/) (even if  that particular scam is less likely here).

The concept of watch-only might become less clear in general moving forward. @sipa does this get in the way of the direction you have in mind for the wallet refactor?

**Comment by instagibbs** [comment]
@fanquake based on your image, I think some indications that private keys are disabled is probably desired? Otherwise it looks to a long-time user that indeed you own the funds and have the keys locally.

**Comment by Sjors** [comment]
I believe (didn't test) that you still see the eye icon next to transactions to indicate watch-only, but I think a regular icon in the bottom right would be better. We could get rid of the HD icon to keep visual clutter constant.

File: src/qt/bitcoingui.h [hunk 3]
```diff
@@ -215,7 +215,7 @@ public Q_SLOTS:
      @param[in] status            current hd enabled status
```
Comment (Sjors) [inline src/qt/bitcoingui.h]: Nit: missing param comment

**Review by Sjors — APPROVED** [review]
tACK 82d6c5a on macOS 10.13.6

Before:
before

After:
after

**Review by meshcollider — COMMENTED** [review]
Tested ACK https://github.com/bitcoin/bitcoin/pull/13966/commits/82d6c5aad330bb06d0c918b4e9304c25ff2bdcc8

As an alternative which I think might be clearer, what about including the eye next to the Balances title like this?

image

micro-nit: typo in first commit message `priveate` -> `private`

**Comment by laanwj** [comment]
utACK 82d6c5aad330bb06d0c918b4e9304c25ff2bdcc8

I agree with @sipa, but a change like that will take some planning on how to avoid disruption for people currently using watch-only in existing wallets.

**Pull request merged**